### PR TITLE
Upgrade Ember CLI and blueprint to 3.22; Drop Node.js 13; add embroid…

### PR DESCRIPTION
**Pull request by SergeAstapov**
Just some updates to the addon/deps/blueprint

### CODE REVIEW

File: addon/scheduler.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { schedule } from '@ember/runloop';
 import { deprecate } from '@ember/debug';
 import Router from '@ember/routing/router';
 import { gte } from 'ember-compatibility-helpers';
-import { buildWaiter, Token } from 'ember-test-waiters';
+import { buildWaiter, Token } from '@ember/test-waiters';
```
Comment (SergeAstapov) [inline addon/scheduler.ts:6]: seem this can happen already?

File: config/ember-try.js [hunk 2]
```diff
@@ -14,6 +14,14 @@ module.exports = async function() {
           },
         },
       },
+      {
```
Comment (SergeAstapov) [inline config/ember-try.js:17]: new LTS, new scenario

File: ember-cli-build.js [hunk 3]
```diff
@@ -13,6 +13,16 @@ module.exports = function(defaults) {
     This build file does *not* influence how the addon or the app using it
     behave. You most likely want to be modifying `./index.js` or app's build file
   */
+  if ('@embroider/webpack' in app.dependencies()) {
```
Comment (SergeAstapov) [inline ember-cli-build.js:16]: Inspired by [Embroider preparation](https://medium.com/@kiwiupover/embroider-preparation-14d59edafc0b) blog post

File: package.json [hunk 4]
```diff
@@ -38,27 +38,27 @@
...
     "@types/ember-testing-helpers": "^0.0.4",
     "@types/ember__test-helpers": "^1.7.0",
     "@typescript-eslint/eslint-plugin": "^3.6.1",
     "@typescript-eslint/parser": "^3.7.0",
     "babel-eslint": "^10.1.0",
     "broccoli-asset-rev": "^3.0.0",
-    "ember-cli": "~3.21.0",
-    "ember-cli-addon-docs": "^0.8.0",
+    "ember-cli": "~3.22.0",
+    "ember-cli-addon-docs": "^0.10.0",
```
Comment (SergeAstapov) [inline package.json:61]: looks like no actual changes, just deps updates https://github.com/ember-learn/ember-cli-addon-docs/blob/master/CHANGELOG.md#v0100-2020-09-08

File: package.json [hunk 5]
```diff
@@ -109,7 +108,7 @@
     "favicons": "5.3.0"
   },
   "engines": {
-    "node": ">= 10.*"
+    "node": "10.* || 12.* || >= 14"
```
Comment (SergeAstapov) [inline package.json:111]: Drop Node.js 13 as it's end of life

File: tests/dummy/app/router.js [hunk 6]
```diff
@@ -1,11 +1,10 @@
 import AddonDocsRouter, { docsRoute } from 'ember-cli-addon-docs/router';
-import RouterScroll from 'ember-router-scroll';
 import config from './config/environment';
 
-const Router = AddonDocsRouter.extend(RouterScroll, {
-  location: config.locationType,
-  rootURL: config.rootURL,
-});
+export default class Router extends AddonDocsRouter {
```
Comment (SergeAstapov) [inline tests/dummy/app/router.js:4]: `AddonDocsRouter` is actually just `class AddonDocsRouter extends EmberRouterScroll {}` per [`ember-cli-addon-docs/addon/router.js`](https://github.com/ember-learn/ember-cli-addon-docs/blob/v0.10.0/addon/router.js) so it's not needed

File: tsconfig.json [hunk 7]
```diff
@@ -1,5 +1,6 @@
 {
   "compilerOptions": {
+    "experimentalDecorators": true,
```
Comment (SergeAstapov) [inline tsconfig.json:3]: not sure if there was specific reason why it wasn't turned on before, WebStorm complained about decorators

**Comment by SergeAstapov** [comment]
`embroider-tests` fail because of ember-cli-addon-docs and it's deps

File: .github/workflows/ci-build.yml [hunk 8]
```diff
@@ -20,7 +20,14 @@ jobs:
...
     steps:
       - if: matrix.os == 'macOS-latest'
+        name: install browsers
+        # MacOS resp. homebrew needs workarounds to make it less unstable.
+        # See https://github.com/actions/virtual-environments/issues/1811#issuecomment-708480190
         run: |
+          brew uninstall openssl@1.0.2t || true
+          brew uninstall python@2.7.17 || true
+          brew untap local/openssl || true
+          brew untap local/python2 || true
```
Comment (rwjblue) [inline .github/workflows/ci-build.yml:30]: I think this can probably be reverted now?

File: addon/scheduler.ts (hunk 1 above)
Comment (rwjblue) [reply addon/scheduler.ts:6]: Ya, we just need to make sure we depend on at least `ember-test-waiters@2.1.0` or `@ember/test-waiters`.

File: package.json (hunk 5 above)
Comment (rwjblue) [reply package.json:111]: Just to call out, this is the main breaking change here.

File: tests/dummy/app/router.js (hunk 6 above)
Comment (rwjblue) [reply tests/dummy/app/router.js:4]: nice!

File: tests/dummy/config/ember-cli-update.json [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+      "version": "3.22.0",
+      "blueprints": [
+        {
+          "name": "addon",
+          "outputRepo": "https://github.com/ember-cli/ember-addon-output",
+          "codemodsSource": "ember-addon-codemods-manifest@1",
+          "isBaseBlueprint": true,
+          "options": [
+            "--yarn"
+          ]
```
Comment (rwjblue) [inline tests/dummy/config/ember-cli-update.json:15]: Should probably also add `--no-welcome`

File: tsconfig.json (hunk 7 above)
Comment (rwjblue) [reply tsconfig.json:3]: Nope, seems good.

**Review by rwjblue — CHANGES_REQUESTED** [review]
This is great, thanks for working on it!

Can you split this up into separate PRs for:

* Adding ember-lts-3.20 and embroider-tests scenarios
* Dropping Node 13
* Updating the rest of the blueprint stuff

That way, we can ensure that things are nicely listed in the changelog and verify things independently.

**Comment by SergeAstapov** [comment]
@rwjblue @scalvert Sorry for delay here.

* Dropping Node.js was moved to #1119
* Embroider setup was moved to #1120

**Pull request closed without merging**